src/libcollections/vec.rs

```diff
@@ -456,8 +458,9 @@ impl<T> Vec<T> {
     }
 
     /// Reserves the minimum capacity for exactly `additional` more elements to
-    /// be inserted in the given `Vec<T>`. Does nothing if the capacity is already
-    /// sufficient.
+    /// be inserted in the given `Vec<T>`. After calling `reserve_exact`,
+    /// capacity will be greater than or equal to `self.len() + additional`.
```

Comment:
@sgrif I had presumed that `precisely minimal` in that context implied `capacity can not be relied upon to be precisely describe the backing allocation size`, but you're correct that it isn't entirely clear. Is there a desire to allow `capacity` to (at some point) more accurately expand to the size of the backing memory returned from the allocator? Could allow eliding calling into the allocator in some cases.

@alexcrichton I had taken the `Does nothing if the capacity is already sufficient.` line which immediately follows to create an exception to that. The added sentence could be reworded to make that exception explicit, though.